src/main/webapp/script.js

```diff
@@ -215,6 +221,35 @@ function createCalendar() {
     });
 }
 
+function createGrocery() {
+    const groceryForm = document.getElementById('groceryFormID');
+
+    // creating FormData to get the values of the form
+    const formData = new FormData(groceryForm);
+    var queryString = "";
+    var array = [];
+
+    // loop through the key and values of the form and add them to an array
+    for (var pair of formData.entries()) {
+        var key = pair[0];
+        var value = pair[1]; 
+        queryString += key + '=' + value;
+        array.push(queryString);
+        queryString = "";
+    }
+    document.getElementById('groceryItemID').value='';
+    document.getElementById('assignGroceryID').value='';
+
+    fetch(new Request('/grocery-list', {
+        method: 'POST', body: array.join('&'), 
+        headers: { 
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+    })).then((response) => handleErrors(response)).then((response) => {
```

Comment:
Okay, so looking at this I can't figure out where the weird error from the servlet is coming from either, but I do have a suggestion for how to improve handleErrors to handle this condition.

In handleErrors, please check for the status codes that you're explicitly returning (SC_UNAUTHORIZED, SC_BadRequest), and _only_ use the error message from the response if it's for those status codes. Otherwise, set a generic error message: "Something went wrong. Please try again."

That way, when unexpected errors come up, we will at least have something to display to the user.

@anthonyqzhu FYI